extensions/amp-ad-network-adsense-impl/0.1/amp-ad-network-adsense-impl.js

```diff
@@ -309,9 +338,69 @@ export class AmpAdNetworkAdsenseImpl extends AmpA4A {
   }
 
   /** @override */
+  buildCallback() {
+    super.buildCallback();
+
+    if (this.isResponsive()) {
+      // Attempt to resize to the correct height.
+      const viewport = this.getViewport();
+      this.responsiveSizeChangePromise_ =
+          this.attemptChangeSize(
+              AmpAdNetworkAdsenseImpl.getResponsiveHeightForContext_(
+                  viewport.getSize()),
+              undefined);
+    }
+  }
+
+  /** @override */
+  onLayoutMeasure() {
+    super.onLayoutMeasure();
+
+    if (this.isResponsive() && !this.responsiveAligned_) {
+      this.responsiveAligned_ = true;
+
+      const layoutBox = this.getLayoutBox();
+
+      // Nudge into the correct horizontal position. NB: this must succeed in
+      // order to align the element correctly, so we use changeSize rather than
+      // attemptChangeSize here. But it doesn't actually change the size (it
+      // just changes the horizontal margins).
+      // TODO(charliereams): This is wrong for RTL.
+      this.element.getResources()./*OK*/changeSize(
+          this.element, undefined, undefined, undefined,
+          {left: -1 * layoutBox.left});
+      setStyle(this.element, 'zIndex', 30);
+    }
+  }
+
+  /** @override */
+  layoutCallback() {
+    const callback = super.layoutCallback();
+    if (this.responsiveSizeChangePromise_) {
+      return callback.then(this.responsiveSizeChangePromise_);
```

Comment:
Ah, thanks. No longer necessary to do this at all, it's handled by the promise returned by buildCallback.